s3transfer/download.py

```diff
@@ -518,6 +534,21 @@ def _main(self, client, bucket, key, fileobj, extra_args, callbacks,
                 continue
         raise RetriesExceededError(last_exception)
 
+    def _handle_io(self, download_output_manager, fileobj, chunk, index):
+        download_output_manager.queue_file_io_task(fileobj, chunk, index)
+
+
+class ImmediatelyWriteIOGetObjectTask(GetObjectTask):
+    """GetObjectTask that immediately writes to the provided file object
+
+    This is useful for downloads where it is known only one thread is
+    downloading the object so there is no reason to go through the
+    overhead of using an IO queue and executor.
+    """
+    def _handle_io(self, download_output_manager, fileobj, chunk, index):
+        task = download_output_manager.get_io_write_task(fileobj, chunk, index)
+        task()
```

Comment:
Yes that is exactly what will happen. The exception will be raised and exception set within one of the threads of the request executor.